Tiltfile

```
@@ -387,6 +391,106 @@ def prepare_all():
+        "docker": ["./test/infrastructure/docker/templates"],
+    })
+
+    for provider, provider_dirs in flavor_dirs.items():
+        for flavor_dir in provider_dirs:
+            template_list = [item for item in listdir(flavor_dir)]
+            template_list = [template_path for template_path in template_list if os.path.basename(template_path).endswith("yaml")]
+
+            for template_path in template_list:
+                deploy_templates(template_path, provider, substitutions)
+
+def deploy_templates(template_path, provider, substitutions):
+    # validate template_path exists
+    if not os.path.exists(template_path):
+        fail(template_path + " not found")
+
+    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
+    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.22.6")
+    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
+    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
+
+    basename = os.path.basename(template_path)
+    if basename.endswith(".yaml"):
+        if basename.startswith("clusterclass-"):
+            flavor = basename.replace("clusterclass-", "").replace(".yaml", "")
+            deploy_clusterclass_template(flavor, provider, template_path)
+        elif basename.startswith("cluster-template-"):
+            flavor = basename.replace("cluster-template-", "").replace(".yaml", "")
+            deploy_flavor_template(flavor, provider, template_path)
+
+def deploy_clusterclass_template(flavor, provider, template_path):
+    apply_clusterclass_cmd = "cat " + template_path + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"ClusterClass created from\'" + template_path + "\', don't forget to delete\n\""
+    delete_clusterclass_cmd = kubectl_cmd + " delete clusterclass " + flavor + ' --ignore-not-found=true; echo "\n"'
+
+    local_resource(
+        name = flavor,
+        cmd = apply_clusterclass_cmd,
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = [provider + "-clusterclasses"],
+    )
+
+    cmd_button(
+        flavor + ":apply",
+        argv = ["sh", "-c", apply_clusterclass_cmd],
+        resource = flavor,
+        icon_name = "note_add",
+        text = "Apply ClusterClass",
+    )
+
+    cmd_button(
+        flavor + ":delete",
+        argv = ["sh", "-c", delete_clusterclass_cmd],
+        resource = flavor,
+        icon_name = "delete_forever",
+        text = "Delete ClusterClass",
+    )
+
+def deploy_flavor_template(flavor, provider, template_path):
+    apply_cluster_template_cmd = "CLUSTER_NAME=" + flavor + "-$RANDOM; " + clusterctl_cmd + " generate cluster $CLUSTER_NAME --from " + template_path + " > ./.tiltbuild/" + flavor + "; cat ./.tiltbuild/" + flavor + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"Cluster '$CLUSTER_NAME' created, don't forget to delete\n\""
```

```suggestion
    apply_cluster_template_cmd = "CLUSTER_NAME=" + flavor + "-$RANDOM; " + clusterctl_cmd + " generate cluster $CLUSTER_NAME --from " + template_path + " > ./.tiltbuild/" + flavor + "; cat ./.tiltbuild/" + flavor + " | " + kubectl_cmd + " apply -f - && echo \"Cluster '$CLUSTER_NAME' created, don't forget to delete\n\""
```
clusterctl already does envsubst internally, we shouldn't do it twice